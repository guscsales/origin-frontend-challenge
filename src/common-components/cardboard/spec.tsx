import * as React from 'react';
import { mount } from 'enzyme';
import theme from '../../configs/styles/themes';
import { ThemeProvider } from 'styled-components';
import Cardboard from '.';
import { create } from 'react-test-renderer';

describe('Cardboard', () => {
  const getWrapper = (props: any = {}, renderType: any = mount) =>
    renderType(
      <ThemeProvider theme={theme}>
        <Cardboard {...props}>{props.children}</Cardboard>
      </ThemeProvider>
    );

  describe('#render', () => {
    it('should match a snapshot', () => {
      const wrapper = getWrapper({ children: 'test' }, create).toJSON();

      expect(wrapper).toMatchSnapshot();
    });

    it('should return a `Card`', () => {
      const wrapper = getWrapper({ children: 'test' });

      expect(wrapper.exists()).toBe(true);
      expect(wrapper.find('Card')).toHaveLength(1);
    });
  });
});
