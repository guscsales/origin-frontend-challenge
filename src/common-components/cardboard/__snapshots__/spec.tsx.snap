// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cardboard #render should match a snapshot 1`] = `
.c0 {
  -webkit-animation: fadeIn 800ms ease-out;
  animation: fadeIn 800ms ease-out;
}

.c1 {
  background: #fff;
  border-radius: 7px;
  border: 1px solid #e1e8ed;
}

@media (min-width:1024px) {
  .c0 {
    width: 56rem;
    margin: 0 auto;
  }
}

@media (min-width:320px) {
  .c1 {
    box-shadow: 0 0 3px rgba(0,0,0,0.02);
    height: 100%;
    border-radius: 10px 10px 0 0;
    border: none;
  }
}

@media (min-width:1024px) {
  .c1 {
    height: auto;
    border: 1px solid #e1e8ed;
  }
}

@media (min-width:320px) {
  .c2 {
    padding: 2rem 2.5rem;
  }
}

@media (min-width:1024px) {
  .c2 {
    padding: 4rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      test
    </div>
  </div>
</div>
`;
